pubmed_parser/pubmed_oa_parser.py

```diff
@@ -161,13 +161,20 @@ def parse_pubmed_xml(path, include_path=False, nxml=False):
         journal = ""
 
     dict_article_meta = parse_article_meta(tree)
-    pub_year_node = tree.find(".//pub-date/year")
+    pub_year_node = tree.find(".//pub-date[@pub-type='ppub']/year")
     pub_year = pub_year_node.text if pub_year_node is not None else ""
-    pub_month_node = tree.find(".//pub-date/month")
+    pub_month_node = tree.find(".//pub-date[@pub-type='ppub']/month")
     pub_month = pub_month_node.text if pub_month_node is not None else "01"
-    pub_day_node = tree.find(".//pub-date/day")
+    pub_day_node = tree.find(".//pub-date[@pub-type='ppub']/day")
     pub_day = pub_day_node.text if pub_day_node is not None else "01"
 
+    pub_year_node = tree.find(".//pub-date[@pub-type='epub']/year")
+    epub_year = pub_year_node.text if pub_year_node is not None else ""
+    pub_month_node = tree.find(".//pub-date[@pub-type='epub']/month")
+    epub_month = pub_month_node.text if pub_month_node is not None else "01"
+    pub_day_node = tree.find(".//pub-date[@pub-type='epub']/day")
+    epub_day = pub_day_node.text if pub_day_node is not None else "01"
```

Comment:
Since this code does the same twice, would you mind creating a new function for it, internal (same script) or elsewhere?